bangc-ops/mlu_op.h

```diff
@@ -3776,13 +3776,266 @@ mluOpStatus_t MLUOP_WIN_API mluOpFill(
+ * RoiawarePool3dForward operation.
+ *
+ * @par Return
+ * - ::MLUOP_STATUS_SUCCESS, ::MLUOP_STATUS_BAD_PARAM
+ *
+ * @par Scale Limitation
+ * - None.
+ *
+ * @par Requirements
+ * - None.
+ *
+ * @par Example
+ * - None.
+ *
+ * @par Reference
+ * - None.
+ */
+mluOpStatus_t MLUOP_WIN_API
+mluOpGetRoiawarePool3dForwardWorkspaceSize(mluOpHandle_t handle,
+                                           const mluOpTensorDescriptor_t rois_desc,
+                                           const mluOpTensorDescriptor_t pts_desc,
+                                           const mluOpTensorDescriptor_t pts_feature_desc,
+                                           size_t *workspace_size);
+
+// Group:RoiawarePool3d
+/*!
+ * @param[in] handle
+ * Handle to an MLUOP context that is used to manage MLU devices and queues in
+ * ::mluOpRoiawarePool3dForward operation. For detailed information, see ::mluOpHandle_t.
+ * @param[in] pool_method
+ * Pooling method of Roiaware, 0 is 'maxpool', 1 is 'avgpool'. Default: 0.
+ * @param[in] boxes_num
+ * An integer value which is the number of the rois.
+ * @param[in] pts_num
+ * An integer value which is the number of the pts.
+ * @param[in] channels
+ * An integer value which is the number of the pts feature of channels.
+ * @param[in] rois_desc
+ * Descriptor of rois tensor, containing the dimension and layout of the rois tensor.
+ * @param[in] rois
+ * Pointer to the MLU memory that stores the ROI tensor.
+ * @param[in] pts_desc
+ * Descriptor of pts tensor, containing the dimension and layout of the pts tensor.
+ * @param[in] pts
+ * Pointer to the MLU memory that stores the pts tensor.
+ * @param[in] pts_feature_desc
+ * Descriptor of pts_feature tensor, containing the dimension and layout of the pts_feature tensor.
+ * @param[in] pts_feature
+ * Pointer to the MLU memory that stores the pts_feature tensor.
+ * @param[in] workspace
+ * Pointer to the MLU memory that is used as an extra workspace for the
+ * RoiawarePool3dForward operation.
+ * @param[in] workspace_size
+ * The size of the extra workspace in bytes that needs to be used in
+ * the RoiawarePool3dForward operation. You can get the size of the workspace with
+ * the ::mluOpGetCosineSimilarityWorkspaceSize function.
+ * @param[in] max_pts_each_voxel
+ * The maximum number of points each voxel. An integer value which is the dimension of the pts_idx_of_voxels.
+ * @param[in] out_x
+ * An integer value which is the dimension of the pooled_features.
+ * @param[in] out_y
+ * An integer value which is the dimension of the pooled_features.
+ * @param[in] out_z
+ * An integer value which is the dimension of the pooled_features.
+ * @param[in] argmax_desc
+ * Descriptor of argmax tensor, containing the dimension and layout of argmax tensor.
+ * @param[out] argmax
+ * Pointer to the MLU memory that stores the argmax tensor.
+ * @param[in] pts_idx_of_voxels_desc
+ * Descriptor of pts_idx_of_voxels tensor, containing the dimension of pts_idx_of_voxels tensor.
```

Comment:
The descriptor of pts_idx_of_voxels tensor, which contains the dimension of pts_idx_of_voxels tensor.